piptools/utils.py

```diff
@@ -570,10 +570,21 @@ def override_defaults_from_config_file(
     else:
         config_file = Path(value)
 
-    config = parse_config_file(ctx, config_file)
-    if config:
-        _validate_config(ctx, config)
-        _assign_config_to_cli_context(ctx, config)
+    piptools_config, pipcompile_config, pipsync_config = parse_config_file(
+        ctx, config_file
+    )
+
+    configs = [piptools_config]
+
+    if ctx.command.name == "pip-compile":
+        configs.append(pipcompile_config)
+    elif ctx.command.name == "pip-sync":
+        configs.append(pipsync_config)
+
+    for config in configs:
+        if config:
+            _validate_config(ctx, config)
+            _assign_config_to_cli_context(ctx, config)
```

Comment:
Let's dedent this a bit:
```suggestion
    for config in filter(bool, configs):
        _validate_config(ctx, config)
        _assign_config_to_cli_context(ctx, config)
```